src/bindgen/language_backend/java_jna.rs

```diff
@@ -0,0 +1,860 @@
+                        }
+                    }
+                };
+                write!(out, "{typ}")
+            }
+            Type::Array(ty, _len) => {
+                self.write_type(out, ty);
+                out.write("[]");
+            }
+            Type::FuncPtr { .. } => out.write("Callback"),
+        }
+    }
+
+    fn write_documentation<W: Write>(&mut self, out: &mut SourceWriter<W>, d: &Documentation) {
+        if !d.doc_comment.is_empty() {
+            out.new_line_if_not_start();
+            out.write("/**");
+            for line in &d.doc_comment {
+                out.new_line();
+                write!(out, " *{line}")
+            }
+            out.new_line();
+            out.write(" */");
+            out.new_line();
+        }
+    }
+
+    fn write_literal<W: Write>(&mut self, out: &mut SourceWriter<W>, l: &Literal) {
+        match l {
+            Literal::Expr(expr) => {
+                write!(out, "{expr}")
+            }
+            Literal::Struct { export_name, .. } => {
+                // There is an hashmap in there that doesn't have stable debug output
+                not_implemented(&format!("Struct Literal {export_name}"), out)
+            }
+            _ => not_implemented(l, out),
+        }
+    }
+}
+
+enum JnaIntegerType {
+    Byte,
+    Short,
+    Int,
+    NativeLong,
+    Long,
+    SizeT,
+}
+
+impl JnaIntegerType {
+    pub fn size(&self) -> &str {
+        match self {
+            JnaIntegerType::Byte => "1",
+            JnaIntegerType::Short => "2",
+            JnaIntegerType::Int => "4",
+            JnaIntegerType::NativeLong => "Native.LONG_SIZE",
+            JnaIntegerType::Long => "8",
+            JnaIntegerType::SizeT => "Native.SIZE_T_SIZE",
+        }
+    }
+
+    pub fn set_method(&self) -> &str {
+        match self {
+            JnaIntegerType::Byte => "setByte(0, (byte)value.intValue())",
+            JnaIntegerType::Short => "setShort(0, (short)value.intValue())",
+            JnaIntegerType::Int => "setInt(0, value.intValue())",
+            JnaIntegerType::NativeLong | JnaIntegerType::SizeT => {
+                "setNativeLong(0, new NativeLong(value.longValue()))"
+            }
+            JnaIntegerType::Long => "setLong(0, value.longValue())",
+        }
+    }
+
+    pub fn get_method(&self) -> &str {
+        match self {
+            JnaIntegerType::Byte => "getByte(0)",
+            JnaIntegerType::Short => "getShort(0)",
+            JnaIntegerType::Int => "getInt(0)",
+            JnaIntegerType::NativeLong | JnaIntegerType::SizeT => "getNativeLong(0).longValue()",
+            JnaIntegerType::Long => "getLong(0)",
+        }
+    }
+
+    pub fn from_kind(kind: &IntKind) -> Self {
+        match kind {
+            IntKind::Short => JnaIntegerType::Short,
+            IntKind::Int => JnaIntegerType::Int,
+            IntKind::Long => JnaIntegerType::NativeLong,
+            IntKind::LongLong => JnaIntegerType::Long,
```

Comment:
LongLong is not Long, for sure.

Reply:
Long is 64 bytes in java (the C long is NativeLong)